### PR TITLE
Observable: add the ability to pass params to handlers.

### PR DESCRIPTION
**Pull request by sriramkrish85**
Part 2 for #4292

**Yet to come**

- [ ]  Enable Instrumentation.triggerEvent to accept object params (and observable to send params)
- [ ]  Have carousel send carousel-id and data-slide-id to analytics

### CODE REVIEW

File: src/observable.js [hunk 1]
```diff
@@ -60,13 +60,16 @@ export class Observable {
...
    * @param {TYPE=} opt_event
+   *
    */
-  fire(opt_event) {
+  fire(opt_event, var_args) {
     const handlers = this.handlers_;
     for (let i = 0; i < handlers.length; i++) {
       const handler = handlers[i];
-      handler(opt_event);
+      handler.apply(null, arguments);
```
Comment (jridgewell) [inline src/observable.js]: This will be a massive slowdown.

Comment (sriramkrish85) [inline src/observable.js]: agreed will do this only when we have var_args.

Comment (sriramkrish85) [inline src/observable.js]: Done

File: src/observable.js [hunk 2]
```diff
@@ -69,7 +69,11 @@ export class Observable {
     const handlers = this.handlers_;
     for (let i = 0; i < handlers.length; i++) {
       const handler = handlers[i];
-      handler.apply(null, arguments);
+      if (var_args) {
```
Comment (jridgewell) [inline src/observable.js]: `var_args` might be falsey. You'd have to check beforehand:

```js
const apply = arguments.length > 1;
for (...) {
  if (apply) {
    //...
  } else {
    //...
  }
}

Comment (sriramkrish85) [inline src/observable.js]: Good catch!

**Comment by sriramkrish85** [comment]
reconsidering this.

**Pull request closed without merging**